src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseAPI.php

```diff
@@ -0,0 +1,426 @@
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $requestId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_CLAIM_REQUEST_ID);
+        if ($this->getClaimService()->getClaimDao()->getClaimRequestById($requestId) == null) {
+            throw new InvalidParamException();
+        }
+        $claimExpenses = $this->getClaimService()->getClaimDao()->getClaimExpenseList($requestId);
+        $count = $this->getClaimService()->getClaimDao()->getClaimExpenseCount($requestId);
+        $total = $this->getClaimService()->getClaimDao()->getClaimExpenseTotal($requestId);
+        return new EndpointCollectionResult(ClaimExpenseModel::class, $claimExpenses, new ParameterBag([CommonParams::PARAMETER_TOTAL => $count, self::PARAMETER_TOTAL_AMOUNT => $total]));
+    }
+
+    /**
+     * @return ParamRuleCollection
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->RequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_CLAIM_REQUEST_ID,
+                    new Rule(Rules::POSITIVE)
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_REQUEST_ID,
+                    new Rule(Rules::POSITIVE)
+                )
+            ),
+        );
+    }
+
+    /**
+     * @OA\Post(
+     *     path="/api/v2/claim/expenses",
+     *     tags={"Claim/Expenses"},
+     *     @OA\RequestBody(
+     *         @OA\JsonContent(
+     *             type="object",
+     *             @OA\Property(property="expenseTypeId", type="integer"),
+     *             @OA\Property(property="amount", type="float"),
+     *             @OA\Property(property="requestId", type="integer"),
+     *             @OA\Property(property="note", type="string"),
+     *             @OA\Property(property="date", type="datetime"),
+     *             required={"name", "expenseTypeId", "amount", "requestId", "date"}
+     *         )
+     *     ),
+     *     @OA\Response(response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 ref="#/components/schemas/Claim-ExpenseModel"
+     *             ),
+     *             @OA\Property(property="meta", type="object")
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $claimExpense = new ClaimExpense();
+        $this->addClaimRequest($claimExpense);
+        $this->setClaimExpense($claimExpense);
```

Comment:
wrong. two transactions handle separately. transaction should come to here `create` method